gui: Add rgb side bar dimming duration control, tool light brightness control and display backlight brightness control

bkerler opened the pull request:
This adds a control for the side bar led dimming duration in seconds, adds a brightness control for the tool led and a display backlight brightness control in percentage.
Related to issue #4139 and #3813. Somewhat related to #660.

bbqsrc commented:
Anything to be done for this to be mergeable?

bkerler commented:
Yep, devs have to review it :)

andy9a9 commented:
What's the status here? :) I would really like an option on CoreOne to dimm the LCD backlight.

bkerler committed:
gui: Add configurable side LED dimming duration

Replace the hardcoded 120-second inactivity timeout with a user-
configurable setting (0–3600 s, default 120 s, stored in config store).

- SideStripHandler: move active_timeout_ms from a static constexpr to
  a runtime member, loaded from config store in load_config(); expose
  get_dimming_timeout_ms() / set_dimming_timeout() accessors.
- store_definition: add side_leds_dimming_duration StoreItem.
- MI_SIDE_LEDS_DIMMING_DURATION: new WiSpin menu item in the Lights
  Settings menu, disabled when dimming is set to Never.

bkerler committed:
gui: Add configurable display backlight brightness

Allow the user to set the LCD backlight brightness (2–100%, default
100%) via the Lights Settings menu. The value is persisted in config
store and restored on boot.

- store_definition: add leds_display_backlight_brightness StoreItem.
- LEDManager::init(): initialise LCD brightness from config store
  instead of the hardcoded 100% constant.
- MI_DISPLAY_BACKLIGHT_BRIGHTNESS: new WiSpin menu item shown for all
  HAS_LEDS() printers in the Lights Settings menu.

bkerler committed:
gui: Add tool light brightness control for toolchanger printers

Allow the user to set the tool LED brightness (0–100%, default 100%)
independently of the on/off toggle. The value is persisted in config
store and applied whenever the LEDs are enabled.

- store_definition: add tool_leds_brightness StoreItem.
- Dwarf::set_cheese_led(): scale brightness from config store instead
  of always using full power (0xff) when LEDs are enabled.
- MI_TOOL_LEDS_BRIGHTNESS: new WiSpin menu item shown for toolchanger
  printers in the Lights Settings menu.

bkerler commented:
Just updated to reflect the latest master

shipped-it commented:
Progress? This is ready, no?
